src/services/nifSemanticValidationService.js

```diff
@@ -0,0 +1,100 @@
+services.factory('nifSemanticValidationService', [nifSemanticValidationService]); 
+
+function nifSemanticValidationService() {
+	
+	function isDNISemanticallyCorrect(dni) {
+		return isDNIControlDigitCorrect(dni);
+	}
+
+	function isNIESemanticallyCorrect(nie) {
+		var firstNIELetter = nie.charAt(0).toUpperCase();
+		var firstNIENumber;
+
+		if (firstNIELetter == 'X') {
+			firstNIENumber = '0';
+		} else if (firstNIELetter == 'Y') {
+			firstNIENumber = '1';
+		} else if (firstNIELetter == 'Z') {
+			firstNIENumber = '2';
+		}
+
+		var otherNIEchars = nie.substr(1, nie.length - 1);
+		return isDNIControlDigitCorrect(firstNIENumber + otherNIEchars);
+	}
+
+	function isDNIControlDigitCorrect(dni) {
+		var possibleControlDigits = 'TRWAGMYFPDXBNJZSQVHLCKE';
+
+        var dniNumber = dni.substr(0, dni.length - 1);
+        var dniControlDigit = dni.substr(dni.length - 1, 1).toUpperCase();
```

Comment:
What does make this call? You should enclose this in a separate method and tell what it does